Source/DafnyCore/Resolver/Resolver.cs

```diff
@@ -6427,37 +6427,47 @@ public class ResolveTypeOption {
       var origReporter = this.reporter;
       this.reporter = new ErrorReporterSink();
 
+      var isFailure = ResolveMember(tok, tp, "IsFailure", out _);
+      var propagateFailure = ResolveMember(tok, tp, "PropagateFailure", out _);
+      var extract = ResolveMember(tok, tp, "Extract", out _);
+
       if (hasKeywordToken) {
-        if (ResolveMember(tok, tp, "IsFailure", out _) == null ||
-            (ResolveMember(tok, tp, "Extract", out _) != null) != expectExtract) {
+        if (isFailure == null || (extract != null) != expectExtract) {
           // more details regarding which methods are missing have already been reported by regular resolution
           origReporter.Error(MessageSource.Resolver, tok,
-            "The right-hand side of ':-', which is of type '{0}', with a keyword token must have members 'IsFailure()', {1} 'Extract()'",
+            "The right-hand side of ':-', which is of type '{0}', with a keyword token must have functions 'IsFailure()', {1} 'Extract()'",
             tp, expectExtract ? "and" : "but not");
         }
       } else {
-        if (ResolveMember(tok, tp, "IsFailure", out _) == null ||
-            ResolveMember(tok, tp, "PropagateFailure", out _) == null ||
-            (ResolveMember(tok, tp, "Extract", out _) != null) != expectExtract) {
+        if (isFailure == null || propagateFailure == null || (extract != null) != expectExtract) {
           // more details regarding which methods are missing have already been reported by regular resolution
           origReporter.Error(MessageSource.Resolver, tok,
-            "The right-hand side of ':-', which is of type '{0}', must have members 'IsFailure()', 'PropagateFailure()', {1} 'Extract()'",
+            "The right-hand side of ':-', which is of type '{0}', must have functions 'IsFailure()', 'PropagateFailure()', {1} 'Extract()'",
             tp, expectExtract ? "and" : "but not");
         }
       }
 
+      void checkIsFunction([CanBeNull] MemberDecl memberDecl, bool allowMethod) {
+        if (memberDecl == null || memberDecl is Function) {
+          // fine
+        } else if (allowMethod && memberDecl is Method) {
+          // give a deprecation warning, so we will remove this language feature around the Dafny 4 time frame
+          origReporter.Deprecated(MessageSource.Resolver, tok,
```

Comment:
Youi will need to sprinkle 'ErrorID.None' into these Deprecated calls, here and elsewhere